core/src/main/java/io/debezium/connector/cassandra/CassandraConnectorTaskTemplate.java

```diff
@@ -129,6 +134,11 @@ public CassandraConnectorContext getTaskContext() {
         return taskContext;
     }
 
+    private void initDeserializationModes() {
+        DecimalHandlingMode decimalMode = config.decimalHandlingMode();
+        CassandraTypeDeserializer.setDecimalHandlingMode(decimalMode);
```

Comment:
Would it be possible to pass the decimal mode via `init` method of `CassandraTypeDeserializer`?

Reply:
Commit https://github.com/debezium/debezium-connector-cassandra/pull/78/commits/e8130f9b21534d9639225c02f7c6c43adda80e95 has been pushed to accomplish this. It moves `CassandraTypeDeserializer.init()` from outside `CassandraConnectorTaskTemplate.main()` into the template class. Note that the actual Cassandra deserializer is still defined and passed in from the Cassandra 3/4 wrapper classes. The deserializer needs to be defined outside of the common template class as the upstream `org.apache.cassandra.serializers.TypeSerializer` changed from an interface in Cassandra 3 to a class in Cassandra 4 (https://github.com/apache/cassandra/commit/ccab496d2d37c86341d364dea6c27513fda27331). Moving the deserializer definition itself into the `CassandraConnectorTaskTemplate` class would result in all calls to `CassandraTypeDeserializer.deserializer()` raising the following error when using the Cassandra 4 connector:

```
  java.lang.IncompatibleClassChangeError: Found class org.apache.cassandra.serializers.TypeSerializer, but interface was expected
```